sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -369,16 +371,14 @@ case class FileSourceScanExec(
   }
 
   @transient
-  private lazy val pushedDownFilters = {
-    val supportNestedPredicatePushdown = DataSourceUtils.supportNestedPredicatePushdown(relation)
-    // `dataFilters` should not include any metadata col filters
-    // because the metadata struct has been flatted in FileSourceStrategy
-    // and thus metadata col filters are invalid to be pushed down
-    dataFilters.filterNot(_.references.exists {
-      case FileSourceMetadataAttribute(_) => true
-      case _ => false
-    }).flatMap(DataSourceStrategy.translateFilter(_, supportNestedPredicatePushdown))
-  }
+  private lazy val pushedDownFilters = DataSourceStrategy.getPushedDownDataFilters(
+    dataFilters, relation)
+
+  @transient
+  private lazy val pushedDownRuntimeFilters = DataSourceStrategy.getPushedDownDataFilters(
+    dataFilters.filter(hasScalarSubquery)
+      // Replace ScalarSubquery at runtime, then the filter may be able to push down.
+      .map(_.transform { case s: ExecScalarSubquery => s.value }), relation)
```

Comment:
[doubt] we expect only un-correlated scalar subquery to be here right as they will always be planned / executed before query (filesource scanning starts). But here we are not checking explicitly that ScalarSubQuery is un-correlated. I also don't see the similar checks happening when collecting un-correlated subquery during query execution either .

https://github.com/apache/spark/blob/8dfbb9a94a0d84588ebd341c728159c19123874e/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala#L236-L254


Is this because we assume that the correlated scalar query will be rewritten to left join via `RewriteCorrelatedScalarSubquery`  ?